src/core/meshcop/border_agent.cpp

```diff
@@ -249,6 +249,7 @@ void BorderAgent::HandleCoapResponse(const ForwardContext &aForwardContext,
 BorderAgent::BorderAgent(Instance &aInstance)
     : InstanceLocator(aInstance)
     , mState(kStateStopped)
+    , mPskcChanged(false)
```

Comment:
This is not necessary, because the `Error SecureTransport::SetPsk(const uint8_t *aPsk, uint8_t aPskLength)` is actually setting the key for next session. It doesn't affect ongoing session.